Lib/test/test_tempfile.py

```diff
@@ -1000,6 +1016,21 @@ def test_del_on_close(self):
         finally:
             os.rmdir(dir)
 
+    def test_del_rolled_file(self):
+        # The rolled file should be deleted when the
+        # SpooledTemporaryFile object is deleted. This should raise a
+        # ResourceWarning since the file was not explicitly closed.
+        f = self.do_create(max_size=2)
+        f.write(b'foo')
+        filename = f.name
+        self.assertTrue(os.path.exists(filename))
+        with self.assertWarns(ResourceWarning):
+            f.__del__()
```

Comment:
It is not correct to call `__del__` directly. Instead, use `del f` or `f = None`.